### PR TITLE
Resume sending E2E test failures to slack

### LINKED ISSUE #3555: Spike: report E2E test failures to slack
### Description
The slack alerts for E2E failures stopped working after moving to GH actions from Travis, explore ways to get that working again.

### Acceptance criteria

* Decide whether fixing the slackbot as done in [this PR](https://github.com/Automattic/woocommerce-payments/pull/2347) makes sense or whether we should move to custom slack reporting as done in WC core.
* Implement the decided solution.

### Testing instructions

1. After implementing, have a test E2E run that fails and reports messages to Slack.
2. Check that the reported message has all the required details and screenshot of the failure.

### Additional context

Discussion thread: p1625275687050700-slack-C026J0AL8NS

### PR DESCRIPTION
**Pull request by deepakpathania**
Fixes #3555 

#### Changes proposed in this Pull Request
- Add the required config for using `e2e-environment` to resume sending E2E test failures to slack.
- Rename `jest-setup.js` to `jest.setup.js` and move it to `tests/e2e/config` for it to be picked up correctly.
- Use `useE2EJestConfig` for jest config to use the failure script that includes slack reporting.

#### Testing instructions

- Go to Actions tab ->  WC Pay E2E tests.
- Run the workflow for this branch `update/slack-e2e-reporting`.
- Notice failure messages would be sent to `#wcpay-bots` channel in slack, [reference job](https://github.com/Automattic/woocommerce-payments/actions/runs/1618699792).

Sample slack screenshot

Screenshot 2021-12-24 at 1 12 28 PM

-------------------

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Review by naman03malhotra — COMMENTED** [review]
LGTM, left a comment for my understanding.

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -125,9 +128,12 @@ jobs:
       - name: Run E2E Tests
         run: npm run test:e2e
 
-      # Upload screenshots if any
-      - name: Upload screenshots
+      # Archive screenshots if any
+      - name: Archive e2e test screenshots
+        if: ${{ always() }}
```
Comment (naman03malhotra) [inline .github/workflows/e2e-test.yml:133]: Just to confirm my understanding, we are using [always](https://docs.github.com/en/actions/learn-github-actions/expressions#always) here as this step is independent of previous steps and we'd always want to execute it.

Comment (deepakpathania) [reply .github/workflows/e2e-test.yml:133]: Yes, archiving screenshots should happen regardless of the previous steps' outcome. This is consistent with how we handle the same in other repos that handle the same, [exammple](https://github.com/woocommerce/woocommerce-admin/blob/main/.github/workflows/e2e.yml#L39).

**Review by zmaglica — APPROVED** [review]
LGTM.

**Review by naman03malhotra — APPROVED** [review]

File: .github/workflows/e2e-test.yml [hunk 2]
```diff
@@ -125,9 +128,12 @@ jobs:
...
         uses: actions/upload-artifact@v2
         with:
-          name: screenshots
-          path: screenshots
+            name: e2e-screenshots
+            path: tests/e2e/screenshots
```
Comment (dechov) [inline .github/workflows/e2e-test.yml:137]: Was there a change to where the screenshots get saved? It happens [here](https://github.com/Automattic/woocommerce-payments/blob/63497bd134466478aeb34770b725e52a33cdef39/tests/e2e/utils/helpers.js#L18) but maybe when running locally it depends on where the tests are run from? Not seeing the artifact on recent runs (e.g. [here](https://github.com/Automattic/woocommerce-payments/actions/runs/1755888388)).

Comment (deepakpathania) [reply .github/workflows/e2e-test.yml:137]: @dechov We moved to use the `useE2EJestConfig` provided by `e2e-environment` which sends the failed screenshot messages directly to `#wcpay-bots` channel in slack, which might have overridden the old artifact behavior to only create artifacts for failure cases, see [this run](https://github.com/Automattic/woocommerce-payments/actions/runs/1789277846) for reference.
Do we need the screenshots for success scenarios as well?

Comment (dechov) [reply .github/workflows/e2e-test.yml:137]: Ah, I see 👍

Not that we need the regular ones but there is some theoretical benefit to having snapshots from each distinct screen from the normal run, to ensure that there aren't visual regressions that wouldn't be caught by specific assertions. What do you think of having a "failure screenshots" artifact in addition to the earlier one?

Comment (deepakpathania) [reply .github/workflows/e2e-test.yml:137]: I feel unless there is an alert/checking process in place, we wouldn't really monitor the successful ones regularly, and manually checking visual regressions could be tricky across runs.
We could either ask GS to do some sanity or have explicit visual regression tests in place for the same but that's just my opinion, we could discuss it internally to see what other folks feel about the same.

**Pull request merged**